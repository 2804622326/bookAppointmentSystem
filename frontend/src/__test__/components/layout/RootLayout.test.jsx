import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import '@testing-library/jest-dom';
import RootLayout from '../../../components/layout/RootLayout.jsx';

test('renders layout with outlet', () => {
  const { asFragment } = render(
    <MemoryRouter initialEntries={["/"]}>
      <Routes>
        <Route element={<RootLayout />}> <Route index element={<div>Home</div>} /></Route>
      </Routes>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
