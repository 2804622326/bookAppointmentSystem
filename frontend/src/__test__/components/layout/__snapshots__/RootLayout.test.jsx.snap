// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders layout with outlet 1`] = `
<DocumentFragment>
  <main>
    <nav
      class="nav-bg navbar navbar-expand-lg navbar-light sticky-top"
    >
      <div
        class="container"
      >
        <a
          class="nav-home navbar-brand"
          href="/"
        >
          uniPetcare
        </a>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            class="me-auto navbar-nav"
          >
            <a
              class="nav-link"
              href="/doctors"
            >
              Meet Our Veterinarians
            </a>
          </div>
          <div
            class="navbar-nav"
          >
            <div
              class="nav-item dropdown"
            >
              <a
                aria-expanded="false"
                class="dropdown-toggle nav-link"
                href="#"
                id="basic-nav-dropdown"
                role="button"
                tabindex="0"
              >
                Account
              </a>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="background-slider"
    >
      <div
        class="carousel slide"
      >
        <div
          class="carousel-indicators"
        >
          <button
            aria-current="true"
            aria-label="Slide 1"
            class="active"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 2"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 3"
            data-bs-target=""
            type="button"
          />
        </div>
        <div
          class="carousel-inner"
        >
          <div
            class="active carousel-item"
          >
            <img
              alt="Slide 0"
              class="d-block w-100"
              src="test-file-stub"
            />
          </div>
          <div
            class="carousel-item"
          >
            <img
              alt="Slide 1"
              class="d-block w-100"
              src="test-file-stub"
            />
          </div>
          <div
            class="carousel-item"
          >
            <img
              alt="Slide 2"
              class="d-block w-100"
              src="test-file-stub"
            />
          </div>
        </div>
        <a
          class="carousel-control-prev"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-prev-icon"
          />
          <span
            class="visually-hidden"
          >
            Previous
          </span>
        </a>
        <a
          class="carousel-control-next"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-next-icon"
          />
          <span
            class="visually-hidden"
          >
            Next
          </span>
        </a>
      </div>
    </div>
    <div>
      <div>
        Home
      </div>
    </div>
  </main>
</DocumentFragment>
`;
